lib/api/v1/controllers/result_test.go

```diff
@@ -96,6 +96,131 @@ var _ = Describe("ResultsController", func() {
+				ctx, response := MakeJSONRequest("GET", fmt.Sprintf("/results/%d", result.ID), nil, nil, user)
+				ctx.Params = append(ctx.Params, gin.Param{Key: "id", Value: fmt.Sprint(result.ID)})
+
+				resultsController := controllers.ResultsController{}
+				resultsController.Show(ctx)
+
+				Expect(response.Code).To(Equal(http.StatusOK))
+
+				jsonResponse := &serializers.ResultDetailJSONResponse{}
+				GetJSONResponseBody(response.Result(), &jsonResponse)
+
+				Expect(jsonResponse.Data.ID).To(Equal(fmt.Sprint(result.ID)))
+				Expect(jsonResponse.Data.Relationships.AdLinks.Data).To(HaveLen(1))
+				Expect(jsonResponse.Data.Relationships.AdLinks.Data[0].ID).To(Equal(fmt.Sprint(adLink.ID)))
+				Expect(jsonResponse.Data.Relationships.AdLinks.Data[0].Type).To(Equal("ad_link"))
+				Expect(jsonResponse.Data.Relationships.Links.Data).To(HaveLen(1))
+				Expect(jsonResponse.Data.Relationships.Links.Data[0].ID).To(Equal(fmt.Sprint(link.ID)))
+				Expect(jsonResponse.Data.Relationships.Links.Data[0].Type).To(Equal("link"))
+			})
+		})
+
+		Context("given result ID that does NOT belong to the authenticated user", func() {
+			It("returns not found error", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				anotherUser := FabricateUser(faker.Email(), faker.Password())
+				result := FabricateResult(anotherUser)
+				ctx, response := MakeJSONRequest("GET", fmt.Sprintf("/results/%d", result.ID), nil, nil, user)
+				ctx.Params = append(ctx.Params, gin.Param{Key: "id", Value: fmt.Sprint(result.ID)})
+
+				resultsController := controllers.ResultsController{}
+				resultsController.Show(ctx)
+
+				Expect(response.Code).To(Equal(http.StatusNotFound))
+
+				jsonResponse := &jsonapi.ErrorsPayload{}
+				GetJSONResponseBody(response.Result(), &jsonResponse)
+
+				Expect(jsonResponse.Errors[0].Title).To(Equal(errors.Titles[errors.ErrNotFound]))
+				Expect(jsonResponse.Errors[0].Code).To(Equal(errors.ErrNotFound.Error()))
+				Expect(jsonResponse.Errors[0].Detail).To(Equal("record not found"))
+			})
+		})
+
+		Context("given result ID dose NOT exist", func() {
+			It("returns not found error", func() {
+				user := FabricateUser(faker.Email(), faker.Password())
+				ctx, response := MakeJSONRequest("GET", fmt.Sprintf("/results/%d", 999), nil, nil, user)
+				ctx.Params = append(ctx.Params, gin.Param{Key: "id", Value: "999"})
+
+				resultsController := controllers.ResultsController{}
+				resultsController.Show(ctx)
+
+				Expect(response.Code).To(Equal(http.StatusNotFound))
+
+				jsonResponse := &jsonapi.ErrorsPayload{}
+				GetJSONResponseBody(response.Result(), &jsonResponse)
+
+				Expect(jsonResponse.Errors[0].Title).To(Equal(errors.Titles[errors.ErrNotFound]))
+				Expect(jsonResponse.Errors[0].Code).To(Equal(errors.ErrNotFound.Error()))
+				Expect(jsonResponse.Errors[0].Detail).To(Equal("record not found"))
+			})
+		})
+
+		Context("given INVALID result ID", func() {
+			It("returns not found", func() {
```

Comment:
Should we change this as this action returns `http.StatusBadRequest` and an error? 🤔